### PR TITLE
Infer expect

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
This addresses #150 for ItemGraders only. Implementing this for ItemGrader only was very easy because ItemGraders have a strictly specified answers schema.

I suspect we could do something that works in general, but I would need to think a bit more about it.

I wrote this fairly quickly in case you wanted it for today's workshop. I think it should be safe to merge, but feel free to reject if you want a more general solution =).

### CODE REVIEW

File: mitxgraders/baseclasses.py [hunk 1]
```diff
@@ -98,19 +98,20 @@ def __call__(self, expect, student_input):
         Notes:
             This function ignores the value of expect.
```
Comment (jolyonb) [inline mitxgraders/baseclasses.py]: I think some more rewording is necessary here...

File: mitxgraders/baseclasses.py [hunk 2]
```diff
@@ -98,19 +98,20 @@ def __call__(self, expect, student_input):
...
+            dictionaries to store the expected input along with correctness,
+            grades, and feedback messages.
+
+            Note that authors should still specify the <customresponse />
             expect attribute because its value is displayed to students as
             the "correct" answer.
 
-            The answer that we pass to check is None, indicating that the
-            grader should read the answer from its internal configuration.
+            Note: ItemGraders can infer its answers from the their answers from
```
Comment (jolyonb) [inline mitxgraders/baseclasses.py]: Change to "If no answer is provided in the configuration, an ItemGrader will attempt to infer its answer from the expect parameter of a textline or CustomResponse tag. Note that this doesn't work when an ItemGrader is embedded inside a ListGrader."

**Comment by jolyonb** [comment]
This is exactly what I was after -- I'm a bit surprised it was this simple though! I found that I've been copying code a lot and just changing the "answers" tag, which was getting really annoying. Also, copy paste errors where I put the answer in the expect but forgot to change the answers tag... We should update documentation to reflect this too, but I'm happy with this for the moment.

File: mitxgraders/baseclasses.py (hunk 1 above)
Comment (ChristopherChudzicki) [reply mitxgraders/baseclasses.py]: I just reworded the comments a bit. I left the part about `expect` being ignored: It is true that `AbstractGrader.__call__` completely ignores `expect`. I agree this is a little weird... feel free to word something better :)

**Comment by ChristopherChudzicki** [comment]
> I found that I've been copying code a lot and just changing the "answers" tag, which was getting really annoying

Yeah, I can see that. For all the work I've done on this library, I've probably only written a dozen problems outside the demo course, so I forget about workflow issues sometimes 😦 

> We should update documentation to reflect this too

Yes!

File: mitxgraders/baseclasses.py [hunk 3]
```diff
@@ -92,25 +92,28 @@ def __call__(self, expect, student_input):
...
             expect attribute because its value is displayed to students as
             the "correct" answer.
 
-            The answer that we pass to check is None, indicating that the
-            grader should read the answer from its internal configuration.
+            ItemGraders: If no answer is provided in the configuration, an
+            ItemGrader will attempt to infer its answer from the expect
+            parameter of a textline or CustomResponse tag. Note that this does
+            not work when an ItemGrader is embedded inside a ListGrader. See
+            ItemGrader.__call__ the implementation.
```
Comment (jolyonb) [inline mitxgraders/baseclasses.py]:
```suggestion
            ItemGrader.__call__ for the implementation.
```

**Pull request merged**